include/simics/dmllib.h

```diff
@@ -41,6 +41,22 @@ static inline uint64 DML_shlu(uint64 a, uint64 b)
         return b > 63 ? 0 : a << b;
 }
 
+// This is used instead of SIM_LOG_WARNING directly, as a SIM_get_api_function
+// indirection is needed to work with old 6 base packages
+// TODO scrap this once the base dependency of every package is bumped
+// to 6.0.188 or above.
+#define _DMLLIB_LOG_WARNING(dev, grps, ...) do {                              \
+    void (*log_warning)(conf_object_t *NOTNULL, uint64, const char *NOTNULL,  \
+                        ...)                                                  \
+        = (void (*)(conf_object_t *NOTNULL, uint64, const char *NOTNULL, ...) \
+          )SIM_get_api_function("VT_log_warning");                            \
```

Comment:
Does the `GET_API_FUNCTION` macro help?

Reply:
Oh, interesting, I was not aware of it. It sure would.